### PR TITLE
Remove the js version and update java and windows sdks

### PR DESCRIPTION
**Pull request by ChadKillingsworth**
Removes the GWT transpiled version of Closure-Compiler to support building on newer JDK and Windows SDK versions.

Also allows us to update to newer build images on travis and appveyor

### CODE REVIEW

**Comment by ChadKillingsworth** [comment]
I'd like to get #185 merged first as it makes managing builds in general easier.

**Comment by brad4d** [comment]
@ChadKillingsworth thanks for working on this.

I've added @nreid260 as a reviewer, because I suspect you created this PR as a result of discussions with him.

File: packages/google-closure-compiler-js/readme.md [hunk 1]
```diff
@@ -1,9 +1,5 @@
 # google-closure-compiler-js
 
-JavaScript platform distribution of Closure Compiler.
+**The JS version of closure-compiler is no longer supported or maintained.**
```
Comment (brad4d) [inline packages/google-closure-compiler-js/readme.md:3]: Let's add some explanation for this.

Maybe something like:

> We have decided that the maintenance burden of supporting the JS-compiled version of closure-compiler is greater than its benefits.
> 
> * The need to support compiling Java to JavaScript enforces restrictions on the Java code we can write and our
>   ability to upgrade to newer versions of the JDK.
> * The JDK implemented by J2CL (used to transform Java to JS) has  functional differences that lead to frustrating
>   bugs in the JS version, such as poor support for non-ASCII characters in input sources.
> * There are inevitable differences in the command line flags and options that can be supported by the Java and JS versions of the compiler, leading to user frustration and confusion.
> * Users always have the option to install the JVM and run the Java version.
> * We realize that some users strongly prefer not to install and use a JVM. We believe that these users' needs are satisfied by the Graal compiled-to-native version of the compiler, which is much faster and more functionally identical to the behavior of the Java version than the JS version ever could be.

File: packages/google-closure-compiler-windows/build-image.js [hunk 2]
```diff
@@ -24,7 +24,7 @@
 const fs = require('fs');
 const path = require('path');
 const {DIM, RESET} = require('../../build-scripts/colors');
-const runCommand = require('../../build-scripts/run-command');
+const {spawn} = require('child_process');
```
Comment (brad4d) [inline packages/google-closure-compiler-windows/build-image.js:27]: This seems like an unrelated fixup.

File: packages/google-closure-compiler-js/readme.md (hunk 1 above)
Comment (ChadKillingsworth) [reply packages/google-closure-compiler-js/readme.md:3]: Much better - added.

File: packages/google-closure-compiler-windows/build-image.js (hunk 2 above)
Comment (ChadKillingsworth) [reply packages/google-closure-compiler-windows/build-image.js]: Yeah just missed when I split the github actions PR out.

File: packages/google-closure-compiler-js/readme.md [hunk 3]
```diff
@@ -2,4 +2,17 @@
...
+ - The need to support compiling Java to JavaScript enforces restrictions on the Java code we can write
+ and our ability to upgrade to newer versions of the JDK.
+ - The JDK implemented by J2CL (used to transform Java to JS) has functional differences that lead
+  to frustrating bugs in the JS version, such as poor support for non-ASCII characters in input sources.
+ - There are inevitable differences in the command line flags and options that can be supported by the
+  Java and JS versions of the compiler, leading to user frustration and confusion.
+ - Users always have the option to install the JVM and run the Java version. We realize that some users
+ strongly prefer not to install and use a JVM. We believe that these users' needs are satisfied by the
+ Graal compiled-to-native version of the compiler, which is much faster and more functionally identical
+ to the behavior of the Java version than the JS version ever could be.
```
Comment (nreid260) [inline packages/google-closure-compiler-js/readme.md:17]: Nit: This phrase is strange. "more consistent with"?

File: packages/google-closure-compiler/cli.js [hunk 4]
```diff
@@ -83,116 +83,27 @@ Object.keys(compilerFlags).forEach(flag => {
...
-        delCount++;
-      }
-      args.splice(i, delCount);
-      break;
+const Compiler = require('./lib/node/closure-compiler');
+let args = process.argv.slice(2);
+for (let i = 0; i < args.length; i++) {
+  if (/^--platform/.test(args[i])) {
+    let delCount = 1;
+    if (args[i].indexOf('=') < 0 && args.length > i + 1) {
```
Comment (nreid260) [inline packages/google-closure-compiler/cli.js:91]: Nit: is `includes` avialable?

Comment (ChadKillingsworth) [reply packages/google-closure-compiler/cli.js:91]: Yes - but that would be a change outside this PR. I would like to modernize the code base at some point including utilizing async/await which is also available.

File: packages/google-closure-compiler-js/readme.md (hunk 3 above)
Comment (brad4d) [reply packages/google-closure-compiler-js/readme.md:17]: as a replacement for "functionally identical to"?

Works for me either way.

Comment (ChadKillingsworth) [reply packages/google-closure-compiler-js/readme.md]: I reworded this slightly - see what you think.
